src/blocks/mod.rs

```diff
@@ -301,11 +314,17 @@ impl Block {
                     self.link.as_bytes(),
                 ])
             }
-            _ => todo!(),
+            _ => Err(anyhow!("Block not hashable")),
         };
 
-        let hash = hash_result.with_context(context)?;
-        self.hash = Some(hash);
+        match hash_result {
+            Ok(block_hash) => self.hash = Some(block_hash),
+            Err(error) => {
+                trace!("Ignoring hash for block {:?}. Cause: {}", &self, error);
```

Comment:
This might be useful until we have a better idea how to use `Invalid` and `NotABlock`